vars/projectExists.groovy

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/groovy
+
+def call(Object ctx, String projectName, String token) {
+  def exists = false
+  ctx.openshift.withCluster() {
+    try {
+      if (token) {
+        ctx.openshift.doAs(token) {
```

Comment:
Why?

Reply:
We want to allow you to run this particular pipeline in your own machine or on a cluster that you have access to. Not everyone has access to give the jenkins sa the self-provisioner role which is what you need to check project existence or create new projects. For that case, you can pass your own token in, and we'll use it. For origin-ci, you already granted the jenkins sa the self provisioner role so we should be good without passing one.

Reply:
Makes sense.